src/prefect/agent/kubernetes/agent.py

```diff
@@ -216,6 +226,12 @@ def replace_job_spec_yaml(self, flow_run: GraphQLResult, image: str) -> dict:
             resources["requests"]["cpu"] = os.getenv("JOB_CPU_REQUEST")
         if os.getenv("JOB_CPU_LIMIT"):
             resources["limits"]["cpu"] = os.getenv("JOB_CPU_LIMIT")
+        if self.volume_mounts:
+            job["spec"]["template"]["spec"]["containers"][0][
+                "volumeMounts"
+            ] = self.volume_mounts
+        if self.volumes:
+            job["spec"]["template"]["spec"]["volumes"] = self.volumes
```

Comment:
We should probably add something similar to the imagePullSecrets where we call 
```python
del job["spec"]["template"]["spec"]["imagePullSecrets"]
```
to remove them from the spec if they're empty. Not sure if having empty volume mounts would cause errors on some k8s providers

Reply:
True, that is probably  the safest.